tracer/src/Datadog.Tracer.Native/calltarget_tokens.cpp

```diff
@@ -501,16 +499,28 @@ HRESULT CallTargetTokens::ModifyLocalSig(ILRewriter* reWriter, TypeSignature* me
     *callTargetStateToken = callTargetStateTypeRef;
     *exceptionToken = exTypeRef;
     *callTargetReturnToken = callTargetReturn;
+
+    const auto sizeOfOtherIndexes = indexes.size();
+    auto indexStart = variableNumber + additionalLocalsCount;
+
     if (returnSignatureType != nullptr)
     {
-        *returnValueIndex = newLocalsCount - 4 - additionalLocalsCount;
+        indexStart ++;
+        *returnValueIndex = newLocalsCount - indexStart--;
```

Comment:
Index start is first incremented, then it's used to build `returnValueIndex`, then it's decremented.   I wrote it like that, to make it clear that if there's a return, we want one more index, and then go back to the starting index but I will just change it to indexStart+1 it will be clearer! 😅

Reply:
changed in e905bf5a801e21b9552cdd8316deb9e00a6b50c7

Reply:
Yes right, my code suggestion is not accurate 😅, but yes I wanted to skip the increment-decrement tango.
Resolving.